src/main/java/edu/utdallas/davisbase/storage/IndexPage.java

```diff
@@ -0,0 +1,497 @@
+package edu.utdallas.davisbase.storage;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IndexPage {
+
+	static final int BYTES_OF_PAGE_OFFSET = Short.BYTES;
+	static final int PAGE_OFFSET_OF_CELL_COUNT = 0x01;
+	static final int PAGE_OFFSET_OF_RIGHTMOST_PAGE_NO = 0x06;
+	static final int PAGE_OFFSET_OF_CELL_PAGE_OFFSET_ARRAY = 0x10;
+	static final int PAGE_OFFSET_OF_CELL_CONTENT_START_POINT = 0x03;
+	static final int PAGE_OFFSET_OF_CELL_PARENT_PAGE_NO = 0x0A;
+	static final int pageSize = StorageConfiguration.Builder.getDefaultPageSize();
+	static final int maximumNoOFKeys = 2;
+	static final long metaDataRootPageNoOffsetInFile = 0x05;
+
+	private static int AddInteriorPage(RandomAccessFile file) {
+		int numofPages = 0;
+		try {
+			numofPages = (int) (file.length() / pageSize);
```

Comment:
Narrowing casts are dangerous. If the cast ends up underflowing/overflowing, we'd never know and then the system would be in an invalid state. Either validate the bounds yourself before casting, or use a checked cast instead.

```suggestion
			com.google.common.base.Preconditions.checkState(file.length() / pageSize < Integer.MAX_VALUE, "Cannot increase file size further.");
			numofPages = (int) (file.length() / pageSize);
```